File: Flutter_QuanLyTro/lib/data/models/response/meter_reading_response.dart
class MeterReadingResponse {
  final String? id;
  final String roomNumber;
  final String serviceName;
  final int oldIndex;
  final int newIndex;
  final String readingDate;
  final bool isInvoiced;
  final String? serviceId;

  MeterReadingResponse({
    this.id,
    required this.roomNumber,
    required this.serviceName,
    required this.oldIndex,
    required this.newIndex,
    required this.readingDate,
    required this.isInvoiced,
    this.serviceId,
  });

  factory MeterReadingResponse.fromJson(Map<String, dynamic> json) {
    return MeterReadingResponse(
      id: json['id'] as String?,
      roomNumber: json['roomNumber'] ?? '',
      serviceName: json['serviceName'] ?? '',
      oldIndex: json['oldIndex'] ?? 0,
      newIndex: json['newIndex'] ?? 0,
      readingDate: json['readingDate'] ?? '',
      isInvoiced: json['isInvoiced'] ?? false,
      serviceId: json['serviceId'] as String?,
    );
  }
}

File: BE_QuanLyTro/src/main/java/ut/edu/be_quanlytro/Dto/Response/MeterReadingResponse.java
package ut.edu.be_quanlytro.Dto.Response;

import lombok.Builder;
import lombok.Data;

import java.time.LocalDate;
import java.util.UUID;

@Data
@Builder
public class MeterReadingResponse {
    private UUID id;
    private UUID serviceId;
    private String roomNumber;
    private String serviceName;
    private Integer oldIndex;
    private Integer newIndex;
    private LocalDate readingDate;
    private Boolean isInvoiced;
}